api/routes/grouping.go

```diff
@@ -203,19 +215,17 @@ func createGrouping(dataset string, storageName string, groupingType string, raw
 			return err
 		}
 
-		if tsg.IDCol != "" {
-			// Create a new variable and column for the time series key.
-			if err := task.CreateComposedVariable(meta, data, dataset, storageName, tsg.IDCol, tsg.YCol, tsg.SubIDs); err != nil {
-				return errors.Wrapf(err, "unable to create new variable %s", tsg.IDCol)
-			}
-
-			// Set the name of the expected cluster column - it doesn't necessarily exist.
-			tsg.ClusterCol = model.ClusterVarPrefix + tsg.IDCol
+		// Create a new variable and column for the time series key.
+		if err := task.CreateComposedVariable(meta, data, dataset, storageName, tsg.IDCol, tsg.IDCol, tsg.SubIDs); err != nil {
+			return errors.Wrapf(err, "unable to create new variable %s", tsg.IDCol)
 		}
 
+		// Set the name of the expected cluster column - it doesn't necessarily exist.
+		tsg.ClusterCol = model.ClusterVarPrefix + tsg.IDCol
+
 		// Create a new grouped variable for the time series.
 		groupingVarName := strings.Join([]string{tsg.XCol, tsg.YCol}, task.DefaultSeparator)
-		err = meta.AddGroupedVariable(dataset, groupingVarName, tsg.YCol, model.TimeSeriesType, model.VarDistilRoleGrouping, tsg)
+		err = meta.AddGroupedVariable(dataset, groupingVarName, tsg.YCol, model.TimeSeriesType, model.VarDistilRoleData, tsg)
```

Comment:
Why is this now a data variable and not a grouping variable?

Reply:
The `VarDistilRoleGrouping` role should be applied to a column that is used to group in the case that has to be done.  It shouldn't have been applied to the top level variable.